### PR TITLE
chore: set TS, Eslint, prettier

### LINKED ISSUE #12: Issue RSS-ECOMM-1_08: Configure bundler 🧩
Set up and configure a bundler (e.g., Webpack, Vite, Parcel, Rollup, or other) for the project, supporting both development and production build modes. 💻🔄

Acceptance Criteria 📚
A bundler (e.g., Webpack, Vite, Parcel, Rollup, or other) is set up and integrated with the project. ✔️
The bundler configuration supports both development and production build modes. 🛠️

### LINKED ISSUE #13: Issue RSS-ECOMM-1_09: Configure TypeScript 🖥️
Description 📝
Configure TypeScript to enforce strong typing and improve code quality throughout the project. 💪🏼📚

Acceptance Criteria 📚
TypeScript is configured and integrated into the project. ✔️
The project enforces strong typing and benefits from improved code quality. 📈

### LINKED ISSUE #14: Issue RSS-ECOMM-1_10: Set up ESLint 🖥️
Set up ESLint to enforce consistent coding styles and identify potential issues in the codebase. 🕵️

Acceptance Criteria 📚
ESLint is set up and integrated into the project. ✔️
ESLint enforces consistent coding styles and helps identify potential code issues. 🛠️🧐

### LINKED ISSUE #15: Issue RSS-ECOMM-1_11: Configure Prettier  🎨
Description 📝
Configure Prettier to automatically format code, ensuring a consistent and readable code style. 📚

Acceptance Criteria 📚
Prettier is configured and integrated into the project. ✔️
The project has a consistent and readable code style, thanks to Prettier's automatic formatting. 🖊️

### PR DESCRIPTION
**Pull request by Tetiana-KET**
Resolves #12 , resolves #13, resolves #14, resolves #15 

- [x] Issue RSS-ECOMM-1_08: Set up and configure Vite⚡ for the project, supporting both development and production build modes. 💻🔄 
- [x] Issue RSS-ECOMM-1_09: Configure TypeScript to enforce strong typing and improve code quality throughout the project
- [x] Issue RSS-ECOMM-1_10: Set up ESLint to enforce consistent coding styles and identify potential issues in the codebase.
- [x] Issue RSS-ECOMM-1_11: Configure Prettier to automatically format code, ensuring a consistent and readable code style 🎨

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+		"plugin:@typescript-eslint/recommended",
+		"plugin:react/recommended"
+	],
+	"parser": "@typescript-eslint/parser",
+	"parserOptions": {
+		"ecmaVersion": "latest",
+		"sourceType": "module"
+	},
+	"plugins": ["@typescript-eslint", "react"],
+	"rules": {}
```
Comment (Gromov-Roman) [inline .eslintrc.json:17]: we can add 
```json
"@typescript-eslint/array-type": [
  "error",
  {
      "default": "array-simple"
  }
],
```

Comment (Tetiana-KET) [reply .eslintrc.json:17]: if it is necessary, i will add it ;)

**Review by Maksim99745 — APPROVED** [review]

File: .eslintrc.cjs [hunk 2]
```diff
@@ -3,19 +3,17 @@ module.exports = {
   env: { browser: true, es2020: true },
   extends: [
     'eslint:recommended',
-    'plugin:react/recommended',
```
Comment (Gromov-Roman) [inline .eslintrc.cjs:6]: why removed?

File: .eslintrc.cjs [hunk 3]
```diff
@@ -3,19 +3,17 @@ module.exports = {
...
-    'plugin:react/jsx-runtime',
```
Comment (Gromov-Roman) [inline .eslintrc.cjs:7]: we need tsx plugin

File: index.html [hunk 4]
```diff
@@ -1,13 +1,13 @@
-<!doctype html>
+<!DOCTYPE html>
```
Comment (Gromov-Roman) [inline index.html:1]: If "tabWidth": 2 in prettierrc, why it is 4 here?

Comment (Gromov-Roman) [reply index.html:1]: oh, it is 2 but in tabs

File: .eslintrc.json (hunk 1 above)
Comment (Gromov-Roman) [reply .eslintrc.json:17]:
```json
"no-multiple-empty-lines": [
    "error",
    {
        "max": 1
    }
],
```

Comment (Gromov-Roman) [reply .eslintrc.json:17]:
```json
"no-magic-numbers": ["error", {
    "ignore": [0, 1],
    "ignoreDefaultValues": true,
    "ignoreArrayIndexes": true
}],
```

Comment (Gromov-Roman) [reply .eslintrc.json:17]:
```json
"no-console": [
    "error",
    {
        "allow": [
            "log",
            "warn",
            "dir",
            "timeLog",
            "assert",
            "clear",
            "count",
            "countReset",
            "group",
            "groupEnd",
            "table",
            "dirxml",
            "error",
            "groupCollapsed",
            "Console",
            "profile",
            "profileEnd",
            "timeStamp",
            "context"
        ]
    }
],
```

Comment (Gromov-Roman) [reply .eslintrc.json:17]:
```json
"max-classes-per-file": [
    "error",
    1
],
```

Comment (Gromov-Roman) [reply .eslintrc.json:17]:
```json
"no-debugger": "error",
"lines-between-class-members": ["warn", "always", { "exceptAfterSingleLine": true }],
```

**Pull request closed without merging**